### DISCUSSION TITLE
how to use perpheralCommAdapter

### DISCUSSION THREAD

**Discussion by Just1045523143** [openTCS/opentcs#17] [Q&A]
how to use perpheralCommAdapter-loopback package

**Comment by swltr** [comment]
There is documentation on how to develop peripheral drivers in the Developer's Guide - see https://www.opentcs.org/docs/5.5/developer/developers-guide/opentcs-developers-guide.html#_developing_peripheral_drivers. If you have specific questions regarding that, please make clear what they are.

**Reply by Just1045523143** [reply]
but failed to use it in kernelcontrolcenter,can not choose location's perpheral adapter. all location's adapter is NullPeripheralCommAdapter.

**Reply by swltr** [reply]
Did you add the required property to the location (see https://www.opentcs.org/docs/5.5/user/opentcs-users-guide.html#_adding_and_configuring_peripheral_devices)?
